### ISSUE TITLE
Add the ability to create new users with their own set permissions on the web panel.

### ISSUE THREAD

**Issue by Nightmare-Night** [BlueSquare23/web-lgsm#19]
**Is your feature request related to a problem? Please describe.**
No ability to create new users other than the initial user upon creation.

**Describe the solution you'd like**
Add the ability to create more users for the web panel, and to allow each user to have its own set permissions. These permissions ranging from creating new game servers, deleting game servers, to more specifically start/stopping and editing specific game servers. This could be accessed by the main initial account through the settings page or an additional page.

**Describe alternatives you've considered**

**Additional context**

**Comment by BlueSquare23** [comment]
Yeah good idea! I'll create a todo and update this issue as things progress. Will probably be a significant set of changes and I'll want to think them out carefully. So it might take me a little while. But shouldn't be too hard. I should be able to expand the auth logic to this use case.

Only concern I have is it might break backward compatibility with people upgrading who have an older version of their DB file. But we'll cross that bridge when we come to it...

**Comment by BlueSquare23** [comment]
Still lots and lots to do before this is finished and released but wanted to update the issue to say I've started on this. This new feature is part 2 of 4 of the v1.8.0 release. See [Todos.txt on dev-1.8.0](https://github.com/BlueSquare23/web-lgsm/blob/89d840f848810ba30f183ebce88657f7c750b0d2/Todos.txt) for more info. Will update as things progress.

![image](https://github.com/user-attachments/assets/391a56a8-a05e-462b-8b0d-4c71c8996efb)

**Comment by daerogami** [comment]
As an extension to this, could it be configurable so you can specify a user and the root directory for a given game server? I imagine most people have the default configuration from the guide where they spin up a new user and install the game server in the root of that user's home directory. Would be nice to support that scenario.

A nice-to-have would be an option to make more sophisticated tenancies. Perhaps the web server doesnt even need to live on the host machine. You could give it a list of servers with a sudo user and the web app could manage multiple servers remotely. At least, I am assuming this is currently built to run directly on the host.

**Comment by BlueSquare23** [comment]
Yeah so just following up on your comment from the other day. The plan with this feature is multiple web interface users. These web panel users will be different from the Linux system users on a server. However, yes the app can currently (v1.7) support multiple system users. And the next release (v1.8) will be able to easily run things as other users and even do things on remote machines over SSH through the web panel. So stay tuned! I've been hesitant to give an official release date yet for v1.8, but should be out roughly sometime around Jan 2025.

**Comment by BlueSquare23** [comment]
This is now done! The newest web-lgsm version 1.8.0 is out now! Includes full support for multiple web interface users with configurable permissions! You can add new users, change their passwords, configure their permissions levels etc. all through the web interface now!

Please note the users are web interface only users. They're not related to the linux systems users you have setup on your linux system. 

There are basically three levels of permissions:
1. Original Admin User (cannot be changed, as anti-lockout protection)
2. Other Admin Users (can be created or destroyed same as any other user)
3. Regular Users (have restrict-able permissions, can be limited to individual game servers, commands, pages, etc.)
